### DISCUSSION TITLE
Anonymize or Mask IP

### DISCUSSION THREAD

**Discussion by calluna18** [invertase/react-native-firebase#6232] [General]
Hi!

Is there any way to mask or anonymize IP before sending events to Google? 🙂

**Comment by mikehardy** [comment]
In any privacy-related conversions I think it's important to be extraordinarily specific, so answers have real meaning, and I'm not sure *exactly* what you mean by mask or anonymize IP, or what events are being sent.

That said, my hunch is that no, there is nothing like that possible. Why? Because they don't actually send the IP if I understand correctly, it is not a part of the data payload. It's lower level at the connection between the mobile device and the google servers. You simply cannot establish a connection across the internet to Google without doing it via TCP/IP which strictly requires having an IP address that Google will then be aware of.

The only way to protect privacy then and not have Google know your IP address is to not communicate with Google at all or to restrict by using a VPN, though Firebase is known to sometimes block VPN connections and then fail to work.

On Google's "legal honor" they state that they will use any information collected if you disable collection and there are a variety of toggle variables you may put in firebase.json for this purpose if desired https://rnfirebase.io/app/json-config

**Reply by calluna18** [reply]
Thanks you for your thorough response! 

What happened was that I came across [this article](https://support.google.com/analytics/answer/2763052?hl=en) regarding IP masking for Universal Analytics which made me wonder if there is anything like that in Google Analytics for Firebase or if logging/storing IP is avoided like in Google Analytics 4.

**Reply by mikehardy** [reply]
Ah, I see what you mean. You might like this then :-)

> Google Analytics for Firebase (formerly Firebase Analytics) is now known as Google Analytics 4

https://medium.com/firebase-developers/google-analytics-vs-firebase-analytics-vs-google-analytics-97ca645a8aff

**Reply by calluna18** [reply]
That clarifies things! Thanks a lot! 😃
